findphotodates.py

```
@@ -934,71 +1230,103 @@ def geolocate(lat, lon):
+        if time_since_last < 1.0:
+            time.sleep(1.0 - time_since_last)
 
-            # Fallback to display_name if address components didn't work
-            if not location and "display_name" in data:
-                location = data["display_name"]
+        try:
+            # Build Nominatim reverse geocoding URL with addressdetails=1
+            url = f"https://nominatim.openstreetmap.org/reverse?format=json&lat={lat_float}&lon={lon_float}&addressdetails=1"
+
+            # Create request with proper User-Agent header (required by Nominatim)
+            req = urllib.request.Request(url)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)